go/vt/vtgate/evalengine/comparisons.go

```diff
@@ -28,7 +28,6 @@ type (
 	// when evaluating the whole comparison
 	ComparisonOp interface {
 		Evaluate(left, right EvalResult) (EvalResult, error)
-		IsTrue(left, right EvalResult) (bool, error)
```

Comment:
Why removing `IsTrue`?

Reply:
The reason we had it was that it had lesser memory pressure in the sense that EvalResult was not needed to be created, but once we added comparisons for tuples, we needed to do null safe comparisons on each step leading to the IsTrue calling Evaluate instead of it being the other way around. So there is nothing new that it offers. If we want to use this someplace, then we can add a function for getting a boolean value from the output of the evaluate function